h2o-bindings/bin/custom/R/gen_rulefit.py

```diff
@@ -12,6 +12,57 @@
 parms$ignored_columns <- args$x_ignore
 parms$response_column <- args$y
 """,
+    module="""
+#' Evaluates validity of the given rules on the given data. Returns a frame with a column per each input rule id, 
+#' representing a flag whether given rule is applied to the observation or not.
+#'
+#' @param model A trained rulefit model.  
+#' @param frame A frame on which rule validity is to be evaluated
+#' @param rule_ids Rule ids to be evaluated against the frame
+#' @examples
+#' \dontrun{
+#' library(h2o)
+#' h2o.init()
+#' titanic <- h2o.importFile(
+#'  "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+#' )
+#' response = "survived"
+#' predictors <- c("age", "sibsp", "parch", "fare", "sex", "pclass")
+#' titanic[,response] <- as.factor(titanic[,response])
+#' titanic[,"pclass"] <- as.factor(titanic[,"pclass"])
+#' 
+#' splits <- h2o.splitFrame(data = titanic, ratios = .8, seed = 1234)
+#' train <- splits[[1]]
+#' test <- splits[[2]]
+#' 
+#' rfit <- h2o.rulefit(y = response, x = predictors, training_frame = train, validation_frame = test, 
+#' min_rule_length = 1, max_rule_length = 10, max_num_rules = 100, seed = 1, model_type="rules")
+#' h2o.fit_rules(rfit, train, c("M1T0N7, M1T49N7, M1T16N7", "M1T36N7", "M2T19N19"))
+#' }
+#' @export
+h2o.fit_rules <- function(model, frame, rule_ids) {
+    o <- model
+    if (is(o, "H2OModel")) {
+        if (o@algorithm == "rulefit"){
+            parms <- list()
```

Comment:
Seems like this code could be simplified if we used a rapid expression evaluation

this would avoid the need to write a custom handler and this custom call

Reply:
e3e99b0 + 0a103f7